rpcserver.go

```diff
@@ -1723,9 +1723,11 @@ func (r *rpcServer) SendPaymentSync(ctx context.Context,
 	// largest payment size allotted to (2^32) - 1 mSAT or 4.29 million
 	// satoshis.
 	if amtMSat > maxPaymentMSat {
-		return nil, fmt.Errorf("payment of %v is too large, max payment "+
-			"allowed is %v", nextPayment.Amt,
-			maxPaymentMSat.ToSatoshis())
+		err := fmt.Errorf("payment of %v is too large, max payment "+
+			"allowed is %v", nextPayment.Amt, maxPaymentMSat.ToSatoshis())
+		return &lnrpc.SendResponse{
```

Comment:
What's the reason to do this change? The pattern within the `rpcserver` seems to be returning the actual error on failure.

Reply:
This makes the `SendPaymentSync` behavior match the `SendPayment` async behavior, which returns this type of error as a PaymentError.

Reply:
Fair enough :)